website/docs/r/s3_bucket_versioning.html.markdown

```diff
@@ -0,0 +1,103 @@
+---
+subcategory: "S3"
+layout: "aws"
+page_title: "AWS: aws_s3_bucket_versioning"
+description: |-
+  Provides an S3 bucket versioning resource.
+---
+
+# Resource: aws_s3_bucket_versioning
+
+Provides a resource for controlling versioning on an S3 bucket.
+Deleting this resource will suspend versioning on the associated S3 bucket.
+For more information, see [How S3 versioning works](https://docs.aws.amazon.com/AmazonS3/latest/userguide/manage-versioning-examples.html).
+
+~> **NOTE:** If you are enabling versioning on the bucket for the first time, AWS recommends that you wait for 15 minutes after enabling versioning before issuing write operations (PUT or DELETE) on objects in the bucket.
+
+## Example Usage
+
+```terraform
+resource "aws_s3_bucket" "example" {
+  bucket = "example-bucket"
```

Comment:
`versioning` is already Computed so we shouldn't have to add a `lifecycle` configuration block